### ISSUE TITLE
Does vistion tower trained during starge 2 (Visual Instruction Tuning)?

### ISSUE THREAD

**Issue by GoGoJoestar** [haotian-liu/LLaVA#1537]
I find `@torch.no_grad()` in CLIPVisionTower.forward(), so it won't flow gradient to CLIP while training.

https://github.com/haotian-liu/LLaVA/blob/c121f0432da27facab705978f83c4ada465e46fd/llava/model/multimodal_encoder/clip_encoder.py#L45-L57

However, here is a key: `"mm_vision_tower_lr": 2e-06,"` in model's `config.json` file, and in the [LLaVA-NEXT blog on May 25th](https://llava-vl.github.io/blog/2024-05-25-llava-next-ablations/), the vision tower are training during stage-2 with lr=2e-6. 

Were the previous models trained according to this strategy? Will training CLIP be better when training for downstream task?

**Comment by 2U1** [comment]
I think the training code isn't open yet for the LLaVA-NEXT.

**Comment by PangziZhang523** [comment]
Print out the gradient，while requires_grad=True，the parameter.grad=None?

**Comment by LiZhangMing** [comment]
> Print out the gradient，while requires_grad=True，the parameter.grad=None?

我在vision_tower添加额外的可微调参数，去掉了 @torch.no_grad()，但是在训练过程中requires_grad=True， 而parameter.grad=None。无法更新参数，请问您遇见了这样的问题了吗？怎么解决的呢？有偿回答哈，非常感谢。
